contracts-periphery/src/UniswapWithdrawHook.sol

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+pragma abicoder v2;
+
+import "openzeppelin-contracts/contracts/access/Ownable.sol";
+import "openzeppelin-contracts/contracts/token/ERC20/utils/SafeERC20.sol";
+import "src/interfaces/ISwapRouter.sol";
+
+contract UniswapWithdrawHook is Ownable {
+  using SafeERC20 for IERC20;
+
+  ISwapRouter internal immutable swapRouter;
+  address internal WETH9;
+
+  uint256 public feeBips;
+  address public feeCollector;
+  address payable public feeReceiver;
+
+  constructor(ISwapRouter _swapRouter, address _WETH9, uint256 _feeBips, address payable _feeReceiver) {
+  swapRouter = _swapRouter;
+  WETH9 = _WETH9;
+  feeBips = _feeBips;
+  feeReceiver = _feeReceiver;
+  }
+
+  function tokensWithdrawn(
+    uint256 _amount,
+    address _stealthAddr,
+    address _acceptor,
+    address _tokenAddr,
+    address _sponsor,
+    uint256 _sponsorFee,
+    bytes memory _data
+  ) external{
```

Comment:
I'd recommend doing a formatting check to make things easier to read:
- Missing space before this bracket
- Close parentheses followed by modifier and open parentheses should be on their own line, e.g. L52 below should be changed to

```solidity
  bytes memory _data
) internal {
```

Reply:
`forge fmt` eventually, but until then I just format manually. A regex search for ` \n` is a good way to find trailing spaces

Reply:
Thank you!

Reply:
Should be good now, forgot to save it again after setting up prettier.